mlir/lib/Transforms/lab4/polozov_vladislav/PolozovMaxDepthPass.cpp

```diff
@@ -0,0 +1,60 @@
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/IR/Region.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Tools/Plugins/PassPlugin.h"
+#include <map>
+using namespace mlir;
+
+namespace {
+class PolozovMaxDepthPass
+    : public PassWrapper<PolozovMaxDepthPass, OperationPass<func::FuncOp>> {
+public:
+  StringRef getArgument() const final { return "max-depth-pass"; }
+  StringRef getDescription() const final {
+    return "Counts the max depth of region nests in the function.";
```

Comment:
I suppose that it counts the depth in any function, not some concrete one:
```suggestion
    return "Counts the max depth of region nests in a function.";
```